PROTOCOL.md

```diff
@@ -424,6 +429,64 @@ enough to get you started. You can watch the data sent and received from
 the server on a regular connection, or look at the client source code
 for a full list of message types.
 
+**Action requests**
+
+These are how the client sends the player's decisions to the server. Most
+requests are sent with `REQUESTID`. `REQUESTID` is used to identify which
+action the request was intended for and is used to protect against race
+conditions with the cancel button. The first decision of a match has a
+`REQUESTID` of 1, and each successful decision increments the counter by 1.
+`REQUESTID` is optional.
+
+`|/team ORDER|REQUESTID`
+
+> Chooses the team order. Numbers not listed are displaced to the back by swapping
+> them with the number that took their place. For example `/team 25` sets the team
+> order from default to 253416.
+
+`|/move NUMBER TARGET|REQUESTID`
+
+> Requests to use the selected move `NUMBER` on `TARGET` Pokemon. `NUMBER` starts at
+> 1 and can be up to 12 (though usually 4), where `NUMBER` calls the move indexed at
+> `NUMBER - 1` in the Pokemon's set.
+>
+> Moves with `TARGET` specify which position they are trying to use the move on as a
+> number wherein the opposing Pokemon are positive integers counting up from `1`
+> starting on the right. Ally Pokemon targets are negative integers counting down from
+> `-1` starting on the left. `TARGET` is optional and only needs to be specified for
+> single target moves in doubles/triples formats.
+
+`|/switch NUMBER|REQUESTID`
+
+> Requests to switch the active Pokemon with the Pokemon at position `NUMBER` on the
+> team. In cases where a Pokemon is KO'd, their replacement is chosen with `/switch`.
+
+`|/choose ACTION,ACTION,ACTION|REQUESTID`
+
+> For doubles/triples formats, decisions are sent for all team positions in the same
+> line separated by commas. `ACTION` can be any of the following: `move`, `switch`,
+> `shift`, `pass`.
+>
+> `move` and `switch` use the same syntax as their respective commands explained above.
+> In triples `/choose shift` requests to `|swap|` the current outside Pokemon to the
+> middle team position. When a team position is empty from the remaining number of 
+> Pokemon on the team being fewer than the number of active team positions for the
+> format, `pass` is used to indicate the empty slot is not preforming an action. `pass`
+> is also used when switching in Pokemon to replace fainted ones to indicate the active
+> Pokemon that do not need to be replaced. For example, `/choose move 1 2,move 4 -1,pass`
+> will have the left-most Pokemon attack the opponent's middle Pokemon with it's first
+> move, the middle Pokemon will attack it's ally to the left with with it's fourth move,
+> and the third team slot is empty, presumably because you're bad and attacking your own
+> Pokemon.
+>
+> If an insufficient number of actions or invalid actions are sent, the game will replace
+> the missing or erroneous request with the first valid choice, which is usually the
+> first non-disabled move.
```

Comment:
"first usable move, or Struggle if there are no usable moves left."

It's important because running out of PP is also a possibility, and it's clearer if you say "usable" instead of "non-disabled".